sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala

```diff
@@ -131,9 +131,22 @@ abstract class EventSourcedEntityRouter[S, E <: EventSourcedEntity[S]](protected
         val snapshot =
           if (shouldSnapshot) Option(endState)
           else None
-        CommandResult(events.toVector, commandEffect.secondaryEffect(endState), snapshot, currentSequence)
+
+        try {
+          context.asInstanceOf[CommandContextImpl].checkActive()
```

Comment:
As this is the `EventSourcedEntityRouter` we know it has been created with an instance of `CommandContextImpl`. Not too sure though if this is the right approach.

Reply:
We shouldn't check active though but rather toggle it to active, no such method right now (but only add that if needed and you see it fail without this `checkActive()` call, I think we may be fine without it)